components/esp_modem/src/esp_modem_uart.cpp

```diff
@@ -160,6 +160,11 @@ void UartTerminal::task()
                 ESP_LOGW(TAG, "unknown uart event type: %d", event.type);
                 break;
             }
+        } else {
+            uart_get_buffered_data_len(uart.port, &len);
+            if (len && on_read) {
+                on_read(nullptr, len);
```

Comment:
If I understand correctly, you are calling the read callback with null buffer to consume the data, it wouldn't be better to handle the situation locally?

Reply:
It's more convenient this way, since the terminal class doesn't have to **own** any buffer. we just leave it to upper layers to use the buffer *they own* for reading. (plus practical consideration: if we handle this locally some bytes may theoretically come after reading locally and could be missed in the actual callback)